CMakeLists.txt

```diff
@@ -47,18 +47,21 @@ add_definitions('-g')
 add_definitions('-Wall')
 
 set(CMAKE_MODULE_PATH ${PROJECT_SOURCE_DIR}/cmake-modules)
+include_directories(SYSTEM ${LIBP11_INCLUDE_DIR})
 
 # find all required libraries
 set(Boost_USE_STATIC_LIBS ON)
 find_package(Boost COMPONENTS filesystem iostreams program_options system REQUIRED)
 find_package(CURL REQUIRED)
 find_package(PkgConfig REQUIRED)
+find_package(LibP11 REQUIRED)
+find_package(OpenSSL REQUIRED)
 pkg_search_module(GLIB REQUIRED glib-2.0)
 
 # Use C++11, but without GNU or other extensions
 set(CMAKE_CXX_STANDARD 11)
 
-add_executable(lmp-device-register src/main.cpp)
-target_include_directories(lmp-device-register PRIVATE ${GLIB_INCLUDE_DIRS})
-target_link_libraries(lmp-device-register ${CURL_LIBRARIES} ${Boost_LIBRARIES} ${GLIB_LDFLAGS})
+add_executable(lmp-device-register src/main.cpp src/options.cpp src/auth.cpp src/pkcs11.cpp src/openssl.cpp)
+target_include_directories(lmp-device-register PRIVATE $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/inc> ${GLIB_INCLUDE_DIRS})
+target_link_libraries(lmp-device-register ${CURL_LIBRARIES} ${Boost_LIBRARIES} ${GLIB_LDFLAGS} ${OPENSSL_LIBRARIES} ${LIBP11_LIBRARIES})
 install(TARGETS lmp-device-register RUNTIME DESTINATION bin)
```

Comment:
We can use the libp11.pc pkgconfig provide in the project. With this we can remove the `cmake-modules/FindLibP11.cmake`
```
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -46,17 +46,15 @@ message(STATUS "Setting SOTA_CLIENT to ${SOTA_CLIENT}")
 add_definitions('-g')
 add_definitions('-Wall')
 
-set(CMAKE_MODULE_PATH ${PROJECT_SOURCE_DIR}/cmake-modules)
-include_directories(SYSTEM ${LIBP11_INCLUDE_DIR})
-
 # find all required libraries
 set(Boost_USE_STATIC_LIBS ON)
 find_package(Boost COMPONENTS filesystem iostreams program_options system REQUIRED)
 find_package(CURL REQUIRED)
+find_package(OpenSSL 3.0.0 REQUIRED)
+
 find_package(PkgConfig REQUIRED)
-find_package(LibP11 REQUIRED)
-find_package(OpenSSL REQUIRED)
 pkg_search_module(GLIB REQUIRED glib-2.0)
+pkg_check_modules(LIBP11 REQUIRED libp11)
 
 # Use C++11, but without GNU or other extensions
 set(CMAKE_CXX_STANDARD 11)
```

Reply:
that would be great. let me test this.